ls1intum/Artemis
# show number of open Complaints for Exercises in Tutor dashboard

### Checklist

### Motivation and Context


Resolve #586 
### Description

I added the property numberOfOpenComplaints, which is the number of unevaluated complaints for an Exercise. And the property numberOfOpenMoreFeedbackRequests, which is the number of unevaluated more feedback requests. The sum of these values is then displayed in the tutor participation graph in the form of a progress bar. Calculation: the sum of evaluated complaints and feedback requests divided by the sum of total nr of complaints and feedback requests.
In addition, the definition of the "show finished exercises" was modified. Exercises that still have open complaints or feedback requests are not finished exercises even if all submissions were assessed.

### Steps for Testing


1. Navigate to "Course Administration"
2. Click on "Tutor Course Dashboard" for a course of your selection
a List of unfinished exercises should be displayed with the newly added progress bar (in the tutor participation graph) 
3. Click on the Checkbox "show finished exercises"
exercises that neither have open assessments nor complaints and feedback will be added to the list of displayed exercises
4. Click on "Exercises Dashboard" the newly added progress bar indicating the evaluation of complaints and feedback progress should be displayed again in the tutor participation graph

### Screenshots


![image](https://user-images.githubusercontent.com/50635005/72206461-73bd2f00-348e-11ea-9efe-f3eafbb1a32b.png)
![image](https://user-images.githubusercontent.com/50635005/72206469-86cfff00-348e-11ea-8e42-23aac359439c.png)
![image](https://user-images.githubusercontent.com/50635005/72206479-b1ba5300-348e-11ea-9081-b38997f16a09.png)
